simulator/infra/instrcache/instr_cache.h

```diff
@@ -13,16 +13,19 @@
 #include <utility>
 
 #include <infra/types.h>
+#include <infra/replacement/cache_replacement.h>
 
 template <typename Key, typename Value, size_t CAPACITY>
-class LRUCache
+class InstrCache
 {
         struct Deleter;
     public:
-        LRUCache()
+        InstrCache(int replacement_policy = LRU)
```

Comment:
I think there is no need to have anything other than LRU here.
That cache is not the "physical" cache (i.e. it does not model the real hardware) so we can use ideal replacement policy,